src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -126,7 +126,7 @@ public function __construct(ParameterBagInterface $parameterBag = null)
         parent::__construct($parameterBag);
 
         $this->trackResources = interface_exists('Symfony\Component\Config\Resource\ResourceInterface');
-        $this->setDefinition('service_container', (new Definition(ContainerInterface::class))->setSynthetic(true));
+        $this->setDefinition('service_container', (new Definition(parent::class))->setSynthetic(true)->setPublic(false));
```

Comment:
should we qualify using `parent::class` a bugfix?, given https://github.com/ro0NL/symfony/blob/c3e5afa75c820eb4cad0fecc6ddb40055b75a389/src/Symfony/Component/DependencyInjection/ContainerBuilder.php#L1171